[HLSL][DirectX] Implement WaveGetLaneCount lowering

Keenuts opened the pull request:
Fixes #99159

Keenuts committed:
[HLSL][DirectX] Finish WaveGetLaneCount implementation

Some testing was lacking, as well as DirectX lowering of the intrinsic.

Fixes #99159

farzonl reviewed (approved):
lgtm

spall commented on clang/test/SemaHLSL/WaveGetLaneCount-sm6.hlsl:
```diff
@@ -0,0 +1,9 @@
+// RUN: %clang_cc1 -finclude-default-header -triple dxil-pc-shadermodel6.0-library -verify %s
```
is this test needed if the codegen test is changed to sm 6.0?

bogner committed:
Merge branch 'main' into fix-99159

bogner committed:
Remove stale comment

bogner committed:
Merge branch 'main' into fix-99159

bogner committed:
Delete clang/test/SemaHLSL/WaveGetLaneCount-sm5.hlsl

This test isn't really that valuable - we test availability attributes elsewhere.

bogner committed:
Delete clang/test/SemaHLSL/WaveGetLaneCount-sm6.hlsl

This test isn't really that valuable - we test availability attributes elsewhere.
